aws-blog-event-driven-batch-analytics/README.md

```diff
@@ -0,0 +1,164 @@
+
+### Getting Started
+
+1. Create S3 bucket. This is the bucket where you would maintain data and configurations for testing the sample implementation provided here
+
+ ```
+  aws s3 mb <<S3_EDBA_BUCKET>>
+  ```
+2. Create  [Amazon RDS Mysql 5.7.x instance](http://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/CHAP_GettingStarted.CreatingConnecting.MySQL.html)
+3. Update [lambda function properties](src/main/resources/edba_lambda_config.properties) with your MySQL endpoint, username and password
+4. Create EMR cluster with tag "edba=true". When submitting aggregation jobs to the cluster the "Aggregation Job Submission” layer lambda function will look for the active clusters that have this tag. If you wish to use to a different tag, update [lambda function properties](src/main/resources/edba_lambda_config.properties) to reflect the same. If you plan to use an existing cluster with similar configuration below, just [add the tag](http://docs.aws.amazon.com/ElasticMapReduce/latest/DeveloperGuide/emr-plan-tags-add.html) "edba=true"
+
+  ```
+    aws emr create-cluster --name “MY_EDBA_CLUSTER" --release-label emr-5.0.0 --use-default-roles --ec2-attributes KeyName=my-key --applications Name=Hadoop Name=Spark --region my-region --instance-groups InstanceGroupType=MASTER,InstanceCount=1,InstanceType=m3.xlarge InstanceGroupType=CORE,InstanceCount=3,InstanceType=m3.xlarge  --tags edba=true
+  ```
+5. Update [job configurations](resources/edba_config_mysql.sql) maintained in state management store with the S3 bucket name you created in step#1 and connect to the mysql database instance through your preferred SQL client to execute sql statements inside resources/edba_config_mysql.sql
+6. Create a two node dc1.large [Redshift cluster](http://docs.aws.amazon.com/redshift/latest/mgmt/managing-clusters-console.html#create-cluster)
+7. Connect to the cluster through your preferred SQL client and execute statements inside resources/edba_redshift.sql file
+8. Update Update [spark job properties](src/main/resources/spark-job.conf) with Redshift cluster endpoint and S3 temporary path
+9. Build the jar by executing maven shade package. You need to execute this command from the directory where the pom.xml is located
+
+  ```
+  mvn package
+  ```
+10. copy the final jar to s3 location you configured in step#3
+
+  ```
+  aws s3 cp ./eventdrivenbatchanalytics.jar s3://<<S3_EDBA_BUCKET>>/code/
+  ```
+11. Create Validation/Conversion Layer Lambda function
+
+  ```
+  aws lambda create-function --function-name validateAndNormalizeInputData --zip-file fileb:///<<MyPath>>/eventdrivenbatchanalytics.jar --handler com.amazonaws.bigdatablog.edba.LambdaContainer::validateAndNormalizeInputData --role arn:aws:iam::<<myAccountNumber>>:role/<<myLambdaRole>> --runtime java8 --timeout 120
```

Comment:
I'd like to leave it as is so that the engineer trying this sample will have an opportunity to maintain configurations and executable in the same location for the Spark aggregation job. Also,  I believe it somehow remove the blackbox feeling by letting the engineer use the jar he built.